nva-datamodel-java/src/main/java/no/unit/nva/model/instancetypes/researchdata/DataManagementPlan.java

```diff
@@ -31,13 +33,33 @@ public class DataManagementPlan implements PublicationInstance<MonographPages> {
      * @param related A collection of URIs referencing things covered by the DMP.
      * @param pages   The pages description for the DMP document.
      */
-    public DataManagementPlan(@JsonProperty(RELATED_FIELD) RelatedUris related,
+    public DataManagementPlan(@JsonProperty(RELATED_FIELD) Set<Object> related,
                               @JsonProperty(PAGES_FIELD) MonographPages pages) {
         this.pages = pages;
-        this.related = nonNull(related) ? new HashSet<>(related) : Collections.emptySet();
+        this.related = nonNull(related)
+                           ? related.stream()
+                                 .map(this::toRelatedDocument)
+                                 .collect(Collectors.toSet())
+                           : Set.of();
+    }
+
+    private RelatedDocument toRelatedDocument(Object item) {
+        if (item instanceof Map map) {
```

Comment:
The logic in these methods can be condensed, but is it worth spending the time? Probably not.